src/renderTopLanguages.js

```diff
@@ -80,15 +109,23 @@ const renderTopLanguages = (topLangs, options = {}) => {
       </style>
       <rect data-testid="card-bg" x="0.5" y="0.5" width="99.7%" height="99%" rx="4.5" fill="${bgColor}" stroke="#E4E2E2"/>
 
-      
+
       ${
         hide_title
           ? ""
           : `<text data-testid="header" x="25" y="35" class="header">Top Languages</text>`
       }
 
       <svg data-testid="lang-items" x="25" y="${hide_title ? 25 : 55}">
-      ${FlexLayout({
+      ${layout === 'compact' ? `
+        ${langs.map((lang, idx) => {
+          const percentage = ((lang.size / totalSize) * (width - 50)).toFixed(2)
+          const output = `<rect data-testid="lang-progress" x='${offset}' y='0' width='${idx === 0 ? parseFloat(+percentage + 5).toFixed(2): percentage}' height='8' fill='${lang.color}' ${idx === 0 || idx === langs.length - 1 ? "rx='5'": ''} />`
+          offset += +percentage
+          return output
+        }).join('')}
+        ${createLanguageTextNode({langs, totalSize, x: 0, y: 25})}
+      ` : FlexLayout({
```

Comment:
This whole part needs to be refactored out of here, it looks too messy. also calculate the values  outside the attributes (makes it prettier)